base/applications/msconfig/autochkpage.c

```diff
@@ -0,0 +1,247 @@
+#include "autochkpage.h"
+
+#include <tchar.h>
+#include <windows.h>
+
+/* VARIABLES *******************************************************************/
+
+HWND hAutoChkPage;
+HWND hAutoChkDialog;
+HKEY hKey;
+LONG Status;
+TCHAR szValueData[255];
+DWORD dwcbData = sizeof(szValueData);
+DWORD dwBufferSize = 0;
+LPWSTR lpsszMultiString = NULL;
+size_t szSize;
+LPWSTR ptr;
+DWORD dwType;
+TCHAR Program[] = _T("System32\\autochk.exe");
+TCHAR SubKey[] = _T("SYSTEM\\CurrentControlSet\\Control\\Session Manager");
+TCHAR ValueName[] = _T("BootExecute");
+TCHAR EnableValue[] = _T("autocheck autochk *\0");
+
+/* FUNCTIONS *******************************************************************/
+
+INT_PTR CALLBACK
+AutoChkPageWndProc(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
+{
+  UNREFERENCED_PARAMETER(lParam);
+  
+  switch (message)
+  {
+    case WM_INITDIALOG:
+    {
+      hAutoChkDialog = hDlg;
+      SetWindowPos(hDlg, NULL, 10, 32, 0, 0, SWP_NOACTIVATE | SWP_NOOWNERZORDER | SWP_NOSIZE | SWP_NOZORDER);
+
+      // Open the key so we can query it
+      Status = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+                            SubKey,
+                            0,
+                            KEY_READ,
+                            &hKey);
+
+      if (Status == ERROR_SUCCESS)
+      {
+        Status = RegQueryValueExW(hKey,
+                                  ValueName,
+                                  NULL,
+                                  NULL,
+                                  (LPBYTE)szValueData,
+                                  &dwcbData);
+
+        if (Status == ERROR_SUCCESS)
+        {
+          // Check the array if it's empty
+          if (szValueData[0] == _T('\0'))
+          {
+            // If the array is empty, always set the Disable radio button
+            SendDlgItemMessage(hDlg, IDC_AUTOCHK_DISABLE, BM_SETCHECK, BST_CHECKED, 0);
+            RegCloseKey(hKey);
+          }
+          else
+          {
+            // Otherwise, always set the Enable radio button
+            SendDlgItemMessage(hDlg, IDC_AUTOCHK_ENABLE, BM_SETCHECK, BST_CHECKED, 0);
+            RegCloseKey(hKey);
+          }
+        }
+        else
+        {
+          // We failed, close the key
+          RegCloseKey(hKey);
+          return FALSE;
+        }
+      }
+      else
+      {
+        // We failed, close the key
+        RegCloseKey(hKey);
+        return FALSE;
+      }
+
+      return TRUE;
+    }
+
+    case WM_COMMAND:
+    {
+      switch ( LOWORD(wParam) )
+      {
+        case IDC_AUTOCHK_ENABLE:
+        {
+
+          // Open the key so we can set its new value
+          Status = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+                                SubKey,
+                                0,
+                                KEY_SET_VALUE,
+                                &hKey);
+
+          if (Status == ERROR_SUCCESS)
+          {
+            // Set autochk as enabled
+            Status = RegSetValueEx(hKey,
```

Comment:
This should not happen on "radio button click", but only once "Apply" or "OK" is clicked.

Reply:
Then what should I do? Use a case like `IDS_OK` or something and put it there?